### PR TITLE
CXX-778 Allow SSL configuration independently of the C driver

### PR DESCRIPTION
**Pull request by acmorrow**

### CODE REVIEW

File: src/mongocxx/config/postlude.hpp [hunk 1]
```diff
@@ -1,3 +1,4 @@
+
```
Comment (xdg) [inline src/mongocxx/config/postlude.hpp]: White space?

**Comment by xdg** [comment]
This seems fine, but I thought the original ticket was about CMake: refuse to build if SSL is requested and libmongoc doesn't support it.

File: src/mongocxx/config/postlude.hpp (hunk 1 above)
Comment (acmorrow) [inline src/mongocxx/config/postlude.hpp]: Oops, fixed thanks.

**Comment by acmorrow** [comment]
@xdg - We could probably do that, but I'm not sure it is worth it. The real point of the original ticket (which I wrote), was to make sure that we could easily test building the C++ driver with and without SSL without needing to build the C driver twice (one with and once without SSL), because at one point we had broken the non-ssl build but didn't notice since the underlying C driver had SSL.

**Comment by acmorrow** [comment]
Ping?

**Comment by xdg** [comment]
I don't object to what you've done. I do think that letting users detect that libmongoc doesn't have SSL support at build time is useful, but I don't think it's a priority.

**Pull request merged**